apiserver/billing/odoostorage/odoo/odoo8/odoo8.go

```diff
@@ -0,0 +1,168 @@
+package odoo8
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	billingv1 "github.com/appuio/control-api/apis/billing/v1"
+	"github.com/appuio/control-api/apiserver/billing/odoostorage/odoo"
+	"github.com/appuio/control-api/apiserver/billing/odoostorage/odoo/odoo8/client"
+	"github.com/appuio/control-api/apiserver/billing/odoostorage/odoo/odoo8/client/model"
+)
+
+// Used to identify the accounting contact of a company.
+const roleAccountCategory = 7
+
+var roleAccountFilter = []any{"category_id", "in", []int{roleAccountCategory}}
+var activeFilter = []any{"active", "in", []bool{true}}
+
+func NewOdoo8Storage(odooURL string, debugTransport bool) odoo.OdooStorage {
+	return &oodo8Storage{
+		sessionCreator: func(ctx context.Context) (client.QueryExecutor, error) {
+			return client.Open(ctx, odooURL, client.ClientOptions{UseDebugLogger: debugTransport})
+		},
+	}
+}
+
+type oodo8Storage struct {
+	sessionCreator func(ctx context.Context) (client.QueryExecutor, error)
+}
+
+func (s *oodo8Storage) Get(ctx context.Context, name string) (*billingv1.BillingEntity, error) {
+	id, err := k8sIDToOdooID(name)
+	if err != nil {
+		return nil, err
+	}
+
+	session, err := s.sessionCreator(ctx)
+	if err != nil {
+		return nil, err
+	}
+	o := model.NewOdoo(session)
+
```

Comment:
I guess if we re-use the session it will eventually timeout and be invalid right? I'm just a bit concerned that we might open a bit too many sessions if something starts to do a lot of get request. But that's probably premature optimization so leaving this is probably fine.

Reply:
I think odoo tools just shows an error 500 after a while. I have not yet implemented session handling since it's rather complicated (we don't have a expiry date AFAIK) and i'm not sure if it's needed.